src/support/generation.c

```diff
@@ -122,6 +148,23 @@ __wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
                 WT_PAUSE();
             else
                 __wt_sleep(0, 10);
+
+            /*
+             * If we wait for more than a minute, log the event. In DIAGNOSTIC mode, abort if we
+             * ever wait more than 3 minutes, that's forever.
+             */
+            if (minutes == 0) {
+                minutes = 1;
+                __wt_seconds(session, &start);
+            } else {
+                __wt_seconds(session, &stop);
+                if (stop - start > minutes * WT_MINUTE) {
+                    WT_IGNORE_RET(__wt_msg(session, "%s generation drain waited %u minutes",
+                      __gen_name(which), minutes));
+                    ++minutes;
+                }
+                WT_ASSERT(session, stop - start < 3 * WT_MINUTE);
```

Comment:
My only comment here is that the WT-5043 debugging fails after 5 minutes, this is 3 minutes. Both of those seem fairly random as "a long time for code but not super long for a person". Both should be the same and use a #define? If you agree, some comments and messages need to change from having the hard-coded number.

Reply:
Since the WT-5043 debugging is going to go away, is there any need to create a shared #define?

Reply:
I don't feel strongly. Just a point for thought. Feel free to merge as-is.